classes/privacy/provider.php

```diff
@@ -0,0 +1,239 @@
+                ts.submission_orcapable,
+                ts.submission_hash,
+                tu.turnitin_uid,
+                tu.instructor_defaults,
+                tu.instructor_rubrics,
+                tu.user_agreement_accepted
+                FROM {context} c
+                INNER JOIN {course_modules} cm ON cm.id = c.instanceid
+                INNER JOIN {turnitintooltwo} t ON t.id = cm.instance
+                LEFT JOIN {turnitintooltwo_submissions} ts ON ts.turnitintooltwoid = t.id
+                LEFT JOIN {turnitintooltwo_users} tu ON ts.userid = tu.userid
+                WHERE c.id {$contextsql}
+                AND ts.userid = :userid
+                ORDER BY cm.id";
+
+        $params = ['userid' => $user->id] + $contextparams;
+
+        $submissions = $DB->get_records_sql($sql, $params);
+        foreach ($submissions as $submission) {
+            $context = \context_module::instance($submission->cmid);
+            self::export_turnitintooltwo_data_for_user((array)$submission, $context, $user);
+        }
+    }
+
+    /**
+     * Export the supplied personal data for a single activity, along with any generic data or area files.
+     *
+     * @param array $submissiondata the personal data to export.
+     * @param \context_module $context the module context.
+     * @param \stdClass $user the user record
+     */
+    protected static function export_turnitintooltwo_data_for_user(array $submissiondata, \context_module $context, \stdClass $user) {
+        // Fetch the generic module data.
+        $contextdata = helper::get_context_data($context, $user);
+
+        // Merge with module data and write it.
+        $contextdata = (object)array_merge((array)$contextdata, $submissiondata);
+        writer::with_context($context)->export_data([], $contextdata);
+
+        // Write generic module intro files.
+        helper::export_context_files($context, $user);
+    }
+
+    /**
+     * Delete all data for all users in the specified context.
+     *
+     * @param \context $context the context to delete in.
+     */
+    public static function delete_data_for_all_users_in_context(\context $context) {
+        global $DB;
+
+        if (empty($context)) {
+            return;
+        }
+
+        if (!$context instanceof \context_module) {
+            return;
+        }
+
+        $instanceid = $DB->get_field('course_modules', 'instance', ['id' => $context->instanceid], MUST_EXIST);
+        $DB->delete_records('turnitintooltwo_submissions', ['turnitintooltwoid' => $instanceid]);
+    }
+
+    /**
+     * Delete all user data for the specified user, in the specified contexts.
+     *
+     * @param approved_contextlist $contextlist a list of contexts approved for deletion.
+     */
+    public static function delete_data_for_user(approved_contextlist $contextlist) {
+        global $DB;
+
+        if (empty($contextlist->count())) {
+            return;
+        }
+
+        $userid = $contextlist->get_user()->id;
```

Comment:
You could just pass `$contextlist->get_user()->id` in below instead of saving it to `$userid`.

Reply:
Done.